### ISSUE TITLE
Negative Prompt support - not found

### ISSUE THREAD

**Issue by zsogitbe** [SciSharp/LLamaSharp#535]
llama.cpp supports negative prompt strings (see main.cpp in Examples), but I cannot find any support here. Haven't I looked well or it does not exist yet?

**Comment by martindevans** [comment]
LLamaSharp doesn't support negative prompts in an easy to use way.

All the necessary low level components are there, but they haven't been put together into an easy to use high level system that allows you to use it by just supplying a string as a negative prompt.

**Comment by zsogitbe** [comment]
I see. This is an important feature since we can control model ethics with it easily. It is probably not a lot of work for someone who knows the code already very well.

**Comment by martindevans** [comment]
Applying the guidance is fairly simple (I've actually just put together the code to do that).

The tricky part is that guidance needs a second entire sequence running in parallel to the first (to generate the guidance logits). This really needs batched execution to be fast, which is supported in LLamaSharp, but only very recently and it doesn't work with any kind of high level executor yet.

**Comment by martindevans** [comment]
@zsogitbe I've implemented a demo for how to use classifier free guidance using the new batched executor. As I mentioned this is not as high level as the existing executors, but it should at least demo how things work at the lower level for now. If you could pull https://github.com/SciSharp/LLamaSharp/pull/536, test it out and leave any feedback on that PR, it would be very appreciated!

**Comment by zsogitbe** [comment]
Added a possible bug issue: https://github.com/ggerganov/llama.cpp/issues/5709
